### PR TITLE
DOCS-688: align configuration instructions across updated components

### PR DESCRIPTION
**Pull request by sguequierre**
* add config tab sub-instruction & create component instructions in Config Builder tab
* change images to markdown across docs
* remove Configuration headings 
* Combine attributes tables & standardize types: JSON only recognizes "number", but "int" is a subset of number-- so I changed int* to int and float* to number

### CODE REVIEW

File: docs/components/motor/28byj-48.md [hunk 1]
```diff
@@ -17,17 +17,14 @@ Viam also supports:
 
 {{< /alert >}}
 
-## Configuration
-
-To configure a `28byj48` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
Comment (npentrel) [inline docs/components/motor/28byj-48.md:22]: This is needed

File: docs/components/motor/28byj-48.md [hunk 2]
```diff
@@ -17,17 +17,14 @@ Viam also supports:
 
 {{< /alert >}}
 
-## Configuration
-
-To configure a `28byj48` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`28byj48`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Comment (npentrel) [inline docs/components/motor/28byj-48.md:25]: I think we should also keep that

File: docs/components/motor/dmc4000.md [hunk 3]
```diff
@@ -15,17 +15,16 @@ When using it, you do not need to configure a [board component](/components/boar
 The DMC-40x0 controller can drive a variety of motor types, but the built-in Viam implementation supports only stepper motors at this time.
 You can drive other types of motors with Viam and the DMC-40x0 controller by [creating a custom motor model](/program/extend/).
 
-## Configuration
-
-Assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`DMC4000`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Comment (npentrel) [inline docs/components/motor/dmc4000.md:21]: I think adding this line near the attributes table is helpful

File: docs/components/motor/gpio/_index.md [hunk 4]
```diff
@@ -12,19 +12,16 @@ The `gpio` model supports [DC motors](https://en.wikipedia.org/wiki/DC_motor) (b
 [Encoders](/components/encoder/) can be configured to work with `gpio` motors.
 Find more information in the [encoded motor documentation](/components/motor/gpio/encoded-motor/).
 
-## Configuration
-
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
Comment (npentrel) [inline docs/components/motor/gpio/_index.md:17]: Need to cover this

File: docs/components/motor/gpio/_index.md [hunk 5]
```diff
@@ -12,19 +12,16 @@ The `gpio` model supports [DC motors](https://en.wikipedia.org/wiki/DC_motor) (b
 [Encoders](/components/encoder/) can be configured to work with `gpio` motors.
 Find more information in the [encoded motor documentation](/components/motor/gpio/encoded-motor/).
 
-## Configuration
-
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`gpio`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Comment (npentrel) [inline docs/components/motor/gpio/_index.md:20]: Near the attributes table would be good

File: docs/components/motor/gpiostepper.md [hunk 6]
```diff
@@ -13,19 +13,16 @@ The `gpiostepper` model of the motor component supports bipolar [stepper motors]
 Viam also supports some more advanced stepper driver chips ([TMC5072](../tmc5072/), [DMC4000](../dmc4000/)) that have their own microcontrollers that handle things like speed and acceleration control.
 {{< /alert >}}
 
-## Configuration
-
-To configure a `gpiostepper` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
Comment (npentrel) [inline docs/components/motor/gpiostepper.md:18]: Needed

File: docs/components/motor/gpiostepper.md [hunk 7]
```diff
@@ -13,19 +13,16 @@ The `gpiostepper` model of the motor component supports bipolar [stepper motors]
 Viam also supports some more advanced stepper driver chips ([TMC5072](../tmc5072/), [DMC4000](../dmc4000/)) that have their own microcontrollers that handle things like speed and acceleration control.
 {{< /alert >}}
 
-## Configuration
-
-To configure a `gpiostepper` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`gpiostepper`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Comment (npentrel) [inline docs/components/motor/gpiostepper.md:21]: Maybe useful near attributes table

File: docs/components/motor/tmc5072.md [hunk 8]
```diff
@@ -11,19 +11,18 @@ This model supports stepper motors controlled by the [TMC5072 chip](https://www.
 
 Whereas a basic low-level stepper driver supported by the [`gpiostepper` model](/components/motor/gpiostepper/) sends power to a stepper motor based on PWM signals from GPIO pins, the TMC5072 chip uses SPI bus to communicate with the board, does some processing on the chip itself, and provides convenient features including StallGuard2<sup>TM</sup>.
 
-## Configuration
-
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
Comment (npentrel) [inline docs/components/motor/tmc5072.md:16]: Needed

File: docs/components/motor/tmc5072.md [hunk 9]
```diff
@@ -11,19 +11,18 @@ This model supports stepper motors controlled by the [TMC5072 chip](https://www.
 
 Whereas a basic low-level stepper driver supported by the [`gpiostepper` model](/components/motor/gpiostepper/) sends power to a stepper motor based on PWM signals from GPIO pins, the TMC5072 chip uses SPI bus to communicate with the board, does some processing on the chip itself, and provides convenient features including StallGuard2<sup>TM</sup>.
 
-## Configuration
-
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`TMC5072`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Comment (npentrel) [inline docs/components/motor/tmc5072.md:19]: Maybe useful near attributes table

**Comment by npentrel** [comment]
We sometimes say "The following attributes are available for ~:" before the tables and sometimes not. We can clean that up in a separate PR though if you want

File: docs/components/motor/28byj-48.md (hunk 1 above)
Comment (sguequierre) [reply docs/components/motor/28byj-48.md:22]: Added back on top!

File: docs/components/motor/28byj-48.md (hunk 2 above)
Comment (sguequierre) [reply docs/components/motor/28byj-48.md:25]: Added 22 & 25 back in on top and attributes, respectively-- 24 is covered

File: docs/components/motor/dmc4000.md (hunk 3 above)
Comment (sguequierre) [reply docs/components/motor/dmc4000.md:21]: Kk, added 21 back in under attributes

File: docs/components/motor/gpio/_index.md (hunk 4 above)
Comment (sguequierre) [reply docs/components/motor/gpio/_index.md:17]: Added!

File: docs/components/motor/gpiostepper.md (hunk 6 above)
Comment (sguequierre) [reply docs/components/motor/gpiostepper.md:18]: Added!

File: docs/components/motor/tmc5072.md (hunk 8 above)
Comment (sguequierre) [reply docs/components/motor/tmc5072.md:16]: Added!

**Comment by sguequierre** [comment]
> We sometimes say "The following attributes are available for ~:" before the tables and sometimes not. We can clean that up in a separate PR though if you want

Tried to cover this across motor when addressing feedback there, might need to make a separate pr/full ticket for everything

**Review by npentrel — APPROVED** [review]

File: docs/components/motor/dmc4000.md [hunk 10]
```diff
@@ -83,24 +82,21 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
...
+| `ticks_per_rotation` | int | **Required** | Number of full steps in a rotation. 200 (equivalent to 1.8 degrees per step) is very common. If your data sheet specifies this in terms of degrees per step, divide 360 by that number to get ticks per rotation. |
+| `serial_path` | string | **Required** | Path to <file>/dev/ttyXXXX</file> file. You can find the `XXXX` by running `dmesg` on the computer plugged into the DMC-40x0. If you leave this empty, Viam will attempt to automatically detect the path. |
+| `controller_axis` | string | **Required** | Physical port label (A-H); select which "axis" the motor is wired to on the controller. |
+| `home_rpm` | int | **Required** | Set speed in revolutions per minute (RPM) that the motor will turn when executing a Home() command (using  DoCommand()). |
+| `dir_flip` | bool | Optional | Flip the direction of the signal sent if there is a DIR pin. |
+| `max_rpm` | int | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
+| `max_acceleration_rpm_per_sec` | int | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```
Comment (sguequierre) [inline docs/components/motor/dmc4000.md:97]:
```suggestion
| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |

```

File: docs/components/motor/dmc4000.md [hunk 11]
```diff
@@ -83,24 +82,21 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
...
+| `low_current` | int | **Required** | Reduce the hold current; [reference here](https://www.galil.com/download/comref/com4103/index.html#low_current_stepper_mode.html). |
+| `ticks_per_rotation` | int | **Required** | Number of full steps in a rotation. 200 (equivalent to 1.8 degrees per step) is very common. If your data sheet specifies this in terms of degrees per step, divide 360 by that number to get ticks per rotation. |
+| `serial_path` | string | **Required** | Path to <file>/dev/ttyXXXX</file> file. You can find the `XXXX` by running `dmesg` on the computer plugged into the DMC-40x0. If you leave this empty, Viam will attempt to automatically detect the path. |
+| `controller_axis` | string | **Required** | Physical port label (A-H); select which "axis" the motor is wired to on the controller. |
+| `home_rpm` | int | **Required** | Set speed in revolutions per minute (RPM) that the motor will turn when executing a Home() command (using  DoCommand()). |
+| `dir_flip` | bool | Optional | Flip the direction of the signal sent if there is a DIR pin. |
+| `max_rpm` | int | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
```
Comment (sguequierre) [inline docs/components/motor/dmc4000.md:96]:
```suggestion
| `max_rpm` | number | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |

```

File: docs/components/motor/dmc4000.md [hunk 12]
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
...
-`amplifier_gain` | int | Set the per phase current (when using stepper amp); see <https://www.galil.com/download/comref/com4103/index.html#amplifier_gain.html>.
-`low_current` | int | Reduce the hold current; see <https://www.galil.com/download/comref/com4103/index.html#low_current_stepper_mode.html>.
-`ticks_per_rotation` | int | Number of full steps in a rotation. 200 (equivalent to 1.8 degrees per step) is very common. If your data sheet specifies this in terms of degrees per step, divide 360 by that number to get ticks per rotation.
-`serial_path` | string | Path to <file>/dev/ttyXXXX</file> file. You can find the `XXXX` by running `dmesg` on the computer plugged into the DMC-40x0. If you leave this empty, Viam will attempt to automatically detect the path.
-`controller_axis` | string | Physical port label (A-H); select which "axis" the motor is wired to on the controller.
-`home_rpm` | integer | Speed in revolutions per minute that the motor will turn when executing a Home() command (using  DoCommand()).
+| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```
Comment (sguequierre) [inline docs/components/motor/dmc4000.md:98]:
```suggestion
| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```

File: docs/components/motor/dmc4000.md [hunk 13]
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
...
+| `low_current` | int | **Required** | Reduce the hold current; [reference here](https://www.galil.com/download/comref/com4103/index.html#low_current_stepper_mode.html). |
+| `ticks_per_rotation` | int | **Required** | Number of full steps in a rotation. 200 (equivalent to 1.8 degrees per step) is very common. If your data sheet specifies this in terms of degrees per step, divide 360 by that number to get ticks per rotation. |
+| `serial_path` | string | **Required** | Path to <file>/dev/ttyXXXX</file> file. You can find the `XXXX` by running `dmesg` on the computer plugged into the DMC-40x0. If you leave this empty, Viam will attempt to automatically detect the path. |
+| `controller_axis` | string | **Required** | Physical port label (A-H); select which "axis" the motor is wired to on the controller. |
+| `home_rpm` | int | **Required** | Set speed in revolutions per minute (RPM) that the motor will turn when executing a Home() command (using  DoCommand()). |
+| `dir_flip` | bool | Optional | Flip the direction of the signal sent if there is a DIR pin. |
+| `max_rpm` | number | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
```
Comment (sguequierre) [inline docs/components/motor/dmc4000.md:96]:
```suggestion
| `max_rpm` | number | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```

File: docs/components/motor/dmc4000.md (hunk 12 above)
Comment (sguequierre) [inline docs/components/motor/dmc4000.md:98]:
``` suggestion
```

File: docs/components/motor/dmc4000.md [hunk 14]
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
...
 
-Name | Type | Description
--------------- | ---- | ---------------
-`amplifier_gain` | int | Set the per phase current (when using stepper amp); see <https://www.galil.com/download/comref/com4103/index.html#amplifier_gain.html>.
-`low_current` | int | Reduce the hold current; see <https://www.galil.com/download/comref/com4103/index.html#low_current_stepper_mode.html>.
-`ticks_per_rotation` | int | Number of full steps in a rotation. 200 (equivalent to 1.8 degrees per step) is very common. If your data sheet specifies this in terms of degrees per step, divide 360 by that number to get ticks per rotation.
-`serial_path` | string | Path to <file>/dev/ttyXXXX</file> file. You can find the `XXXX` by running `dmesg` on the computer plugged into the DMC-40x0. If you leave this empty, Viam will attempt to automatically detect the path.
-`controller_axis` | string | Physical port label (A-H); select which "axis" the motor is wired to on the controller.
-`home_rpm` | integer | Speed in revolutions per minute that the motor will turn when executing a Home() command (using  DoCommand()).
```
Comment (sguequierre) [inline docs/components/motor/dmc4000.md:99]:
```suggestion
```

File: docs/components/motor/dmc4000.md [hunk 15]
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
...
 
-### Optional Attributes
```
Comment (sguequierre) [inline docs/components/motor/dmc4000.md:100]:
```suggestion
```

**Pull request merged**